OrbitBase/include/OrbitBase/Logging.h

```diff
@@ -84,6 +84,29 @@ constexpr const char* kLogTimeFormat = "%Y-%m-%dT%H:%M:%E6S";
   } while (0 && (assertion))
 #endif
 
+// Declare the class inside the macro so that file:line are the ones where SCOPED_TIMED_LOG is used.
+#define SCOPED_TIMED_LOG(format, ...)                                                          \
+  class ScopedTimedLog##__LINE__ {                                                             \
```

Comment:
Having a class name per line seems very weird to me to be honest. I'd prefer having the timer logic in its own file and use a unique local variable name to allow for nested timers. We could probably move down the Timer class from OrbitGl to OrbitBase and modify it slightly so that it could, for example, execute a lambda on destruction that would allow us to "LOG" with the proper line number.

Reply:
Why does it seem weird? Unless there's a specific reason I don't see why make things more complicated.

Reply:
We are essentially replicating an existing timer class inside of a macro, so the code already *is* in OrbitBase, but we can't use it elsewhere. Declaring an entire class every time the macro is used seems weird because there are alternatives that would make the code cleaner, reusable and simply more useful.

Reply:
By the way, ScopedTimedLog##__LINE__ does not do what you think, I'll add concatenation macros

Reply:
https://github.com/google/orbit/pull/1417